manim/mobject/opengl_mobject.py

```diff
@@ -81,7 +81,15 @@ def __init__(
         self.data = getattr(self, "data", {})
         self.uniforms = getattr(self, "uniforms", {})
 
-        self.color = Color(color) if color else None
+        if isinstance(color, list):
+            self.color = []
+            for item in color:
+                if isinstance(item, tuple):
+                    self.color.append(Color(item[0]))
+                else:
+                    self.color.append(Color(item))
+        else:
+            self.color = Color(color) if color else None
```

Comment:
Just to add to the above, the `else` clause keeps everything the way it was originally - which is that you would get an error if you passed in anything other than a color.  What was added here was just logic to parse out the color if you are passing in something compatible with setting the color by z-value.  So the code expands what can be passed in to accommodate setting the color by z-value.

Reply:
What if we make things a bit more explicit: check whether the passed value is a string or already a color first, otherwise check whether it is an iterable (not just a list) whose elements we can then assume to be either colors or tuples, and if thats also not true raise a ValueError?

Reply:
I updated the code to this:

```py
if isinstance(color, (str, Color)):
    self.color = Color(color)
elif isinstance(color, Iterable):   
    self.color = []
    for item in color:
        if isinstance(item, tuple):
            self.color.append(Color(item[0]))
        else:
            self.color.append(Color(item))
else:
    raise ValueError("Invalid color type: " + str(color))
```

The nested `isinstance(item, tuple)` is still looking for a tuple because a string is also an iterable.  Or should I change the nested if/else to check first for a string or color and then iterable and then error if none of those?

Reply:
I was able to get rid of this entirely by using a new argument, `colorscale`.